uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -861,32 +863,56 @@ public static function findPublicHolidayLeaveRequestEvenIfDeleted($contactID, Pu
   }
 
   /**
-   * Returns the leave request days already existing for the given contact
-   * within the $fromDate to $todDate period and having the statuses supplied.
+   * Returns existing leave requests considered "overlapping" for the given contact
+   * within the date/time period, absence type, request type and statuses supplied.
    *
    * @param int $contactID
    * @param string $fromDate
    * @param string $toDate
+   * @param int $typeID
+   * @param string $requestType
    * @param array $leaveRequestStatus
    * @param boolean $excludePublicHolidayLeaveRequests
    *   Whether to exclude public holiday leave requests from overlapping leave requests or not
    *
    * @return \CRM_HRLeaveAndAbsences_BAO_LeaveRequest[]|null
    */
-  public static function findOverlappingLeaveRequests($contactID, $fromDate, $toDate, $leaveRequestStatus = [], $excludePublicHolidayLeaveRequests = true) {
+  public static function findOverlappingLeaveRequests($contactID, $fromDate, $toDate, $typeID, $requestType, $leaveRequestStatus = [], $excludePublicHolidayLeaveRequests = true) {
     $leaveRequestTable = self::getTableName();
     $leaveRequestDateTable = LeaveRequestDate::getTableName();
     $leaveBalanceChangeTable = LeaveBalanceChange::getTableName();
+    $leaveRequestAbsenceType = AbsenceType::findById($typeID);
+    $isCalculationUnitInHours = $leaveRequestAbsenceType->isCalculationUnitInHours();
+    $absenceTypes = AbsenceType::getEnabledAbsenceTypes();
+    $absenceTypesIDsWithSameUnit = [];
+    $absenceTypesIDsWithDifferentUnit = [];
+
+    foreach ($absenceTypes as $absenceType) {
+      if ($leaveRequestAbsenceType->calculation_unit === $absenceType->calculation_unit) {
+        $absenceTypesIDsWithSameUnit[] = $absenceType->id;
+      } else {
+        $absenceTypesIDsWithDifferentUnit[] = $absenceType->id;
+      }
+    }
+
+    $noOverlappingCondition =
+      $isCalculationUnitInHours
+      ?
+        "(lr.from_date < '{$toDate}' AND lr.to_date > '{$fromDate}' AND lr.type_id IN (" . join(',', $absenceTypesIDsWithSameUnit) . "))" .
+        (count($absenceTypesIDsWithDifferentUnit) ? " OR ((lrd.date BETWEEN %3 AND %4) AND lr.type_id IN (" . join(',', $absenceTypesIDsWithDifferentUnit) . "))" : "")
+      : "lrd.date BETWEEN %3 AND %4 AND lr.type_id IN (" . join(',', array_merge($absenceTypesIDsWithSameUnit, $absenceTypesIDsWithDifferentUnit)) . ")";
```

Comment:
Can you use implode instead of join?  The keyword could be confusing as one would think its part of the SQl query.

Reply:
Done.